### DISCUSSION TITLE
UniqueValidator don't respect model error_messages for relation

### DISCUSSION THREAD

**Discussion by kindlycat** [encode/django-rest-framework#7778] [Potential Issue] [answered]
When we have unique relation field (one to one for example) then UniqueValidator should respect error_messages from this field. But for now it always return default error message.

```python
# models
class FirstModel(models.Model):
    pass


class SecondModel(models.Model):
    relation_field = models.OneToOneField(
        'FirstModel',
        on_delete=models.CASCADE,
        error_messages={
            'unique': 'my message'
        },
    )

# serializer
class SecondModelSerializer(serializers.ModelSerializer):
    class Meta:
        model = SecondModel
        fields = ('relation_field',)


# issue
FirstModel.objects.create()
SecondModel.objects.create(relation_field=1)

serializer = SecondModelSerializer(data={'relation_field': 1})
serializer.is_valid()
errors = serializer.errors
```
### Expected
``` errors = {'relation_field': [ErrorDetail(string='my message', code='unique')]}```

### Actual result
``` errors = {'relation_field': [ErrorDetail(string='This field must be unique.', code='unique')]}```

It's the same #2878, but for relation.
I made the PR #7599 to fix this.

**Comment by carltongibson** [comment]
These things are a lot easier to assess in free time if you make the effort to explain the issue fully. Having to dig up the history makes it a bit off putting.

**Reply by kindlycat** [reply]
Sorry for that, I updated description with example.

**Reply by kindlycat** [reply]
Any news? Could you merge it?

**Comment by kindlycat** [comment, marked as answer]
Merged
